library/std/src/sys/unix/process/process_unix.rs

```diff
@@ -340,58 +340,88 @@ impl Command {
         }
 
         unsafe {
-            let mut file_actions = PosixSpawnFileActions(MaybeUninit::uninit());
-            let mut attrs = PosixSpawnattr(MaybeUninit::uninit());
-
-            libc::posix_spawnattr_init(attrs.0.as_mut_ptr());
-            libc::posix_spawn_file_actions_init(file_actions.0.as_mut_ptr());
+            let mut attrs = MaybeUninit::uninit();
+            let attrs = match libc::posix_spawnattr_init(attrs.as_mut_ptr()) {
+                0 => PosixSpawnattr(&mut attrs),
+                e => return Err(io::Error::from_raw_os_error(e)),
+            };
+
+            let mut file_actions = MaybeUninit::uninit();
+            let file_actions = match libc::posix_spawn_file_actions_init(file_actions.as_mut_ptr())
+            {
+                0 => PosixSpawnFileActions(&mut file_actions),
+                e => return Err(io::Error::from_raw_os_error(e)),
+            };
 
             if let Some(fd) = stdio.stdin.fd() {
-                cvt(libc::posix_spawn_file_actions_adddup2(
+                match libc::posix_spawn_file_actions_adddup2(
```

Comment:
Yikes, good thing these calls practically never fail, since `cvt` only checks `-1` to use `errno`, and these return non-zero error numbers directly. But instead of expanding that `match` everywhere, how about adding a new `cvt`-like helper method? Something like `fn cvt_nz(c_int) -> io::Result<()>`.

Reply:
Introduced a local cvt_nz function and used it as appropriate. Quick review suggest that we don't need it anywhere else.